### ISSUE TITLE
Static class members named 'constructor'

### ISSUE THREAD

**Issue by ark120202** [microsoft/TypeScript#33198]
**TypeScript Version:**  3.7.0-dev.20190831

**Search Terms:** class, constructor, static, computed, quoted

**Code**

```ts
class A { static constructor() {} }
class B { static "constructor"() {} }
class C { static ["constructor"]() {} }
```

**Expected behavior:**

Even though it may be confused with non-standard extensions like #265, I think treating all cases equally would be less confusing, because unlike non-static `["constructor"]` there is no difference at runtime.

**Actual behavior:**

`A` and `B` (with #31949) get a `'static' modifier cannot appear on a constructor declaration.
` diagnostic.

**Playground Link:**
http://www.typescriptlang.org/play/#code/MYGwhgzhAECC0G9oQC5hQS2NYB7AdqgE4CuwKuRAFAJSIC+09AUKJDAEKLJqbYBEeQilLlK-WgyatwUaAGFuqdFmgBtQQWJkKRfgF1JCRiyA

**Related Issues:** #31020, https://github.com/eslint/eslint/pull/12110

**Comment by sandersn** [comment]
I don't understand your expected behaviour here. Do you expect an error on all 3 or no errors at all?

As far as I understand, none of these 3 examples declares an actual constructor. Am I correct on that?

**Comment by ark120202** [comment]
> I don't understand your expected behaviour here. Do you expect an error on all 3 or no errors at all?

I think anything consistent would be fine. I personally prefer no errors, since, even if it's a bit confusing, all 3 cases are valid JS and disallowing confusing code better fits linter.

> As far as I understand, none of these 3 examples declares an actual constructor. Am I correct on that?

Yes, all these 3 cases are equivalent to `Class.constructor = function() {}`.
